packages/client/hmi-client/src/components/responsive-matrix/matrix.vue

```diff
@@ -0,0 +1,902 @@
+<template>
+	<div class="matrix-container" ref="matrix_container">
+		<div class="matrix" ref="matrix">
+			<!-- <component
+				v-for="(selectedCell, idx) in selectedCellList"
+				:key="idx"
+				:is="getSelectedDrilldownType(selectedCell)"
+				:getSelectedCellStyle="getSelectedCellStyle"
+				:update="update"
+				:move="move"
+				:selectedCell="selectedCell"
+				:dataRowList="dataRowList"
+				:dataColList="dataColList"
+				:labelRowList="labelRowList"
+				:labelColList="labelColList"
+				:parameters="dataParametersArray"
+				:parametersMin="dataParametersMin"
+				:parametersMax="dataParametersMax"
+				:colorFn="getSelectedGraphColorFn(selectedCell)"
+				@click="selectedCellClick(idx)"
+			/> -->
+		</div>
+	</div>
+</template>
+
+<script lang="ts">
+import { nextTick, PropType } from 'vue';
+
+import chroma from 'chroma-js';
+import { Viewport } from 'pixi-viewport';
+import {
+	Application,
+	DisplayObject,
+	Geometry,
+	Texture,
+	Shader,
+	Mesh,
+	FederatedPointerEvent
+} from 'pixi.js';
+import {
+	SelectedCell,
+	SelectedCellValue,
+	CellStatus,
+	CellType,
+	Uniforms
+} from '@/types/ResponsiveMatrix';
+import { uint32ArrayToRedIntTex } from './pixi-utils';
+
+// import ResponsiveCellBarContainer from './cell-bar-container.vue';
+// import ResponsiveCellLineContainer from './cell-line-container.vue';
+
+import matrixVS from './matrix.vs.glsl';
+import matrixFS from './matrix.fs.glsl';
+// import matrixGridFS from './matrix-grid.fs.glsl';
+
+export default {
+	// ---------------------------------------------------------------------------- //
+	// components                                                                   //
+	// ---------------------------------------------------------------------------- //
+
+	// ///////////////////////////////////////////////////////////////////////////////
+
+	components: {
+		// ResponsiveCellBarContainer,
+		// ResponsiveCellLineContainer
+	},
+
+	// ---------------------------------------------------------------------------- //
+	// props                                                                        //
+	// ---------------------------------------------------------------------------- //
+
+	// ///////////////////////////////////////////////////////////////////////////////
+
+	props: {
+		data: {
+			type: null,
+			default: [[], []] // e.g. [[{}, {}, {}], [{}, {}, {}]]
+		},
+		// labels: {
+		// 	type: null,
+		// 	default: [[], []],
+		// },
+		cellLabelRow: {
+			type: Array as PropType<number[] | string[]>,
+			default() {
+				return [];
+			}
+		},
+		cellLabelCol: {
+			type: Array as PropType<number[] | string[]>,
+			default() {
+				return [];
+			}
+		},
+		fillColorFn: {
+			type: Function,
+			default() {
+				return '#000000';
```

Comment:
We are using [Uncharted Design Tokens](https://github.com/unchartedsoftware/design-tokens/blob/main/build/js/es6/uncharted-tokens.js) they are already installed, so you can just import the colours.

Reply:
I don't think you can use CSS-variables in a pixi/shader context.